File: 448_find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        for (int i = 0; i != nums.size(); ) {
            if (nums[i] != i + 1 && nums[i] != nums[nums[i] - 1])
                swap(nums[i], nums[nums[i] - 1]);
            else
                ++i;
        }
        vector<int> result;
        for (int i = 0; i != nums.size(); ++i)
            if (nums[i] != i + 1)
                result.push_back(i + 1);
        return result;
    }
};

File: 448_find-all-numbers-disappeared-in-an-array.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        for i in xrange(len(nums)):
            index = abs(nums[i]) - 1
            nums[index] = -abs(nums[index])
        return [i + 1 for i in xrange(len(nums)) if nums[i] > 0]
